webapp/channels/src/components/cloud_preview_modal/preview_modal_content.tsx

```diff
@@ -0,0 +1,122 @@
+            embedUrl.searchParams.set('modestbranding', '1'); // Reduce YouTube branding
+            embedUrl.searchParams.set('rel', '0'); // Don't show related videos from other channels
+            embedUrl.searchParams.set('iv_load_policy', '3'); // Hide annotations
+            embedUrl.searchParams.set('cc_load_policy', '0'); // Don't show captions by default
+            embedUrl.searchParams.set('playsinline', '1'); // Play inline on mobile
+            embedUrl.searchParams.set('widget_referrer', typeof window === 'undefined' ? 'https://mattermost.com' : window.location.origin); // Set referrer for analytics
+            embedUrl.searchParams.set('controls', '0');
+
+            // Render YouTube embed
+            return (
+                <iframe
+                    src={embedUrl.toString()}
+                    title={content.title}
+                    width='100%'
+                    height='100%'
+                    frameBorder='0'
+                    allow='accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture'
+                    allowFullScreen={true}
+                    data-testid='youtube-embed'
+                />
+            );
+        }
+
+        // Handle direct video files
+        if (content.videoUrl.endsWith('.mp4') || content.videoUrl.endsWith('.webm') || content.videoUrl.endsWith('.mov')) {
+            return (
+                <video
+                    src={content.videoUrl}
+                    controls={true}
+                    loop={true}
+                    data-testid='video-element'
+                >
+                    <track kind='captions'/>
+                </video>
+            );
+        }
+
+        // Handle images
+        if (content.videoUrl.endsWith('.jpg') || content.videoUrl.endsWith('.jpeg') || content.videoUrl.endsWith('.png') || content.videoUrl.endsWith('.gif') || content.videoUrl.endsWith('.webp')) {
+            return (
+                <img
+                    src={content.videoUrl}
+                    alt={content.title}
+                />
+            );
+        }
+
+        // Fallback for other URLs - treat as image
+        return (
+            <img
+                src={content.videoUrl}
+                alt={content.title}
+            />
+        );
+    };
+
+    return (
+        <div className='preview-modal-content'>
+            {content.skuLabel && (
+                <div className='preview-modal-content__sku-label'>
+                    <MattermostLogo className='preview-modal-content__sku-label-logo'/>
+                    <span>{content.skuLabel}</span>
+                </div>
+            )}
+            <h2 className='preview-modal-content__title'>{content.title}</h2>
+            <div
+                className='preview-modal-content__subtitle'
+                dangerouslySetInnerHTML={{__html: content.subtitle}}
```

Comment:
Should this be using `dangerouslySetInnerHTML`? That seems like a mistake

Reply:
If we're doing that in the future, we should probably look at either iframes embedding content directly from the CWS or using Markdown. Perhaps it's just the prop name that's getting to me, but injecting HTML from somewhere else seems dangerous 😅

Reply:
Yeah in the future if we do go for request-based data I'll definitely pivot to markdown.

Reply:
@nickmisasi what do you mean that you removed it in the next diff? I still see it in the code.

Reply:
@esarafianou it was removed here: https://github.com/mattermost/mattermost/pull/31235/commits/454ba232f360f6ff9f252a3f7268da8766475d33

I don't see any link to it in current code. Could you link?